src/components/enterprise-page/data/hooks.js

```diff
@@ -25,13 +26,18 @@ const defaultBrandingConfig = {
  */
 export function useEnterpriseCustomerConfig(enterpriseSlug) {
   const [enterpriseConfig, setEnterpriseConfig] = useState();
+  let display404Page = false;
 
   useEffect(() => {
     fetchEnterpriseCustomerConfig(enterpriseSlug)
```

Comment:
I feel it might be a bit cleaner if we kept the error and enterpriseConfig separate in the hook.

E.g., if the hook returned the error object itself whenever an error occurs, we could largely use that to determine whether to render `ErrorPage` vs. introducing a `display404Page`. Then, we could let the `NotFoundPage` continue to be rendered when `enterpriseConfig` is defined and `null`. IMO, this feels like a simpler mental model to grasp for this logic. Thoughts?

```jsx
const [enterpriseConfig, fetchError] = useEnterpriseCustomerConfig(enterpriseSlug);

if (fetchError) {
  return <ErrorPage message={fetchError.message} />
}

if (isDefinedAndNull(enterpriseConfig)) {
  return <NotFoundPage />
}
```

Reply:
I was actually thinking it would be better if whether or not to display a 404 page was stateful on the Enterprise page that way any hook could set it but I wasn't sure how to best go about doing that.